edx_when/api.py

```diff
@@ -327,21 +394,14 @@ def set_date_for_block(course_id, block_id, field, date_or_timedelta, user=None,
             existing_date.active = True
         except models.ContentDate.DoesNotExist as error:
             if user:
+                # A UserDate creation below requires an existing ContentDate.
```

Comment:
Thank you for these awesome comments! Always good to have some fresh eyes

Reply:
It took awhile to figure it out! So thought I'd capture it.